public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard/EmailSharingConfiguration.tsx

```diff
@@ -154,7 +154,13 @@ export const EmailSharingConfiguration = () => {
 
   return (
     <form data-testid={selectors.Container} className={styles.container} onSubmit={handleSubmit(onSubmit)}>
-      <Field label="Can view dashboard">
+      <Field
+        label="Can view dashboard"
+        className={cx(
+          { [styles.emailViewOptionsContainer]: watch('shareType') === PublicDashboardShareType.EMAIL },
+          { [styles.linkViewOptionsContainer]: watch('shareType') === PublicDashboardShareType.PUBLIC }
+        )}
```

Comment:
why not add this directly in the Field Component that renders the email input, so it's not necessary to add a condition for CSS. That field is rendered when `watch('shareType') === PublicDashboardShareType.EMAIL ` and will always have a margin-top set. So, if it's PUBLIC, no input and no margin is rendered

Reply:
The `Field` component [always adds a margin-bottom](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Forms/Field.tsx#L48) for itself.
I want the parent (`ConfigPublicDashboard`) to decide the gaps between its elements, so that we can change them all from one place. So I'm removing margin-bottom when the choice is PUBLIC, because it adds 16px we don't need. 

On the other hand, when the choice is EMAIL, then there is another child of the `EmailSharingConfiguration` `<form>` and we need _the distance between those 2 siblings_ to also be `24`, instead of `16` which is the default of the `Field`. 

Now the Field "Invite" needs its default 16px to be removed because it is adding them out of place, same case as when the choice is PUBLIC.

Reply:
mmm I don't like the Field component to add a margin-bottom. I think that's not ok, it should let its parent take that decision 🧐.
Now I see the problem 👍🏻